packages/jsonrpc/src/ws-connection-provider.ts

```diff
@@ -49,7 +49,11 @@ const defaultOptions: Required<WebSocketOptions> = {
   closeMessage: 'Connection was closed, will not reconnect!'
 } as const;
 
-export const webSocketConnection = <TConnection = Connection>(url: string | URL, initOptions?: WebSocketOptions) => {
+export const webSocketConnection = <TConnection = Connection>(
+  url: string | URL,
+  initOptions?: WebSocketOptions,
+  abortSignal?: AbortSignal
```

Comment:
I think I would move this to the options (or initOptions).

Reply:
otherwise you will need to pass something empty if you want to add this abortSignal
`webSocketConnection('url', undefined, abortSignal)`
or
`webSocketConnection('url', {abortSignal})`